src/test/javascript/spec/component/orion/orion-exercise-assessment-dashboard.component.spec.ts

```diff
@@ -63,67 +53,75 @@ describe('OrionExerciseAssessmentDashboardComponent', () => {
+        expect(downloadSubmissionSpy).toHaveBeenCalledTimes(1);
+        expect(downloadSubmissionSpy).toHaveBeenCalledWith(comp.exerciseId, programmingSubmission, 2);
     });
     it('ngOnInit should subscribe correctly', fakeAsync(() => {
         const orionState = { opened: 40, building: false, cloning: false } as OrionState;
         const stateObservable = new BehaviorSubject(orionState);
-        const orionStateStub = stub(orionConnectorService, 'state');
-        orionStateStub.returns(stateObservable);
+        const orionStateStub = jest.spyOn(orionConnectorService, 'state');
+        orionStateStub.mockReturnValue(stateObservable);
 
         const response = of(new HttpResponse({ body: programmingExercise, status: 200 }));
-        const getForTutorsStub = stub(TestBed.inject(ExerciseService), 'getForTutors');
-        getForTutorsStub.returns(response);
+        const getForTutorsStub = jest.spyOn(TestBed.inject(ExerciseService), 'getForTutors');
+        getForTutorsStub.mockReturnValue(response);
 
         comp.ngOnInit();
         tick();
 
-        expect(comp.exerciseId).to.be.equals(10);
-        expect(comp.exercise).to.be.deep.equals(programmingExercise);
-        expect(comp.orionState).to.be.deep.equals(orionState);
+        expect(comp.exerciseId).toBe(10);
+        expect(comp.exercise).toEqual(programmingExercise);
+        expect(comp.orionState).toEqual(orionState);
 
-        expect(getForTutorsStub).to.have.been.calledOnceWithExactly(10);
-        expect(orionStateStub).to.have.been.calledOnceWithExactly();
+        expect(getForTutorsStub).toHaveBeenCalledTimes(1);
+        expect(getForTutorsStub).toHaveBeenCalledWith(10);
+        expect(orionStateStub).toHaveBeenCalledTimes(1);
+        expect(orionStateStub).toHaveBeenCalledWith();
     }));
     it('ngOnInit should deal with error correctly', fakeAsync(() => {
         const orionState = { opened: 40, building: false, cloning: false } as OrionState;
         const stateObservable = new BehaviorSubject(orionState);
-        const orionStateStub = stub(orionConnectorService, 'state');
-        orionStateStub.returns(stateObservable);
+        const orionStateStub = jest.spyOn(orionConnectorService, 'state');
+        orionStateStub.mockReturnValue(stateObservable);
 
         const error = new HttpErrorResponse({ status: 400 });
-        const getForTutorsStub = stub(TestBed.inject(ExerciseService), 'getForTutors');
-        getForTutorsStub.returns(throwError(error));
+        const getForTutorsStub = jest.spyOn(TestBed.inject(ExerciseService), 'getForTutors');
+        getForTutorsStub.mockReturnValue(throwError(error));
 
-        const errorSpy = spy(TestBed.inject(AlertService), 'error');
+        const errorSpy = jest.spyOn(TestBed.inject(AlertService), 'error');
         // counter the initialization in beforeEach
         comp.exercise = undefined as any;
```

Comment:
Why do we need the `as any` here?

Reply:
Because comp.exercise has type Exercise, not Exercise?, so we need to cheat the typesafety here

Reply:
Could you maybe something like "force undefined against typecheck" to document this?